stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -360,7 +363,7 @@ namespace stan {
 
       Eigen::Matrix<double, -1, -1> L_A(value_of_rec(A));
       // NOTE: This should be replaced by some check that comes from a user
-      if (L_A.rows()  > 500) {
+      if (L_A.rows()  > 500 && STAN_GPU) {
```

Comment:
It might be better to explicitly only do this size check if STAN_GPU is defined; I'm not sure I trust the compiler to elide the L_A.rows() call if STAN_GPU is set to 0. Bob or somebody else might know more about the optimizer's behavior there than I do.